docs/developer-guide/infrastructure.md

```diff
@@ -0,0 +1,137 @@
+- Tools don't need reinstalling for each build
+- The tests can be run locally on a developer machine in the same environment
+- Old build environments can be used for binary compatibility, such as CentOS 6 for releases.
+
+To avoid long build queues, commits to `master` and `develop` will be tested with more configurations,
+while PRs and commits to other branches may be tested on a smaller set of platforms. PRs and commits
+to `master` should run a full set of tests on all supported platforms to help ensure that it is always
+ready for a new release. Jobs that run for PRs and all commits are marked with `[commit]`, while jobs
+marked with `[daily]` are only run daily on the `develop` branch. A tag of `[master]` indicates that
+the job is only run for commits and PRs to master.
+
+If more extensive testing is needed for a commit or branch, certain keywords can be included in the
+commit message to trigger additional builds for commits or in the branch name to trigger them for
+a particular branch. At some point in the future, the community server may provide ways to trigger
+extra builds in other circumstances.
+
+Unless otherwise mentioned, jobs will typically generate Java and Python interfaces, run the CI/nightly
+test suite, and run some brief packaging tests.
+
+### Linux
+
+For Linux, typically the jobs run will be the latest gcc and clang releases, and the oldest
+gcc and clang releases supported by HELICS. Unless indicated, they will use a system install of ZeroMQ,
+build all supported core types, and use a fairly recent version of dependencies.
+
+- GCC 7.0 on Ubuntu 18.04 using default apt-get versions of dependencies
+- Clang 5.0 using minimum supported version of all dependencies
+- GCC 10 [commit]
+- Clang 10 [commit]
+- GCC 10 with ZeroMQ as a subproject [daily]
+
+For auto-generated swig PRs, there is a special build that will run to test a build using the
+pre-generated swig interface files.
+
+### Windows
+
+Most of the Windows tests run frequently are using MSVC, though MinGW, MSYS, and Cygwin jobs are run daily.
+The MINGW/MSYS builds can be triggered by including `mingw`, `msys`, and/or `cygwin` in a commit message or
+branch name.
+
+- MSVC2017 32bit Build and test
+- MSVC2017 64bit Build and test with Java and Python
+- MSVC2019 64bit Build and test with Java and Python [commit]
+- MinGW [daily]
+- MSYS [daily]
+- Cygwin [daily]
+
+### macOS
+
+macOS builds and tests tend to take a long time to run, so there are only a few of them, and they
+only run on PRs unless `xcode10` or `xcode11` is included in a commit message or branch name.
+In general, the XCode jobs will be the most recent major XCode version and the oldest XCode version
+released in about the past 3 years that's supported by Apple (or is available on the Azure macOS
+images). Apple tends to be pretty good at getting people to upgrade and dropping support for older
+releases.
+
+- XCode 10
```

Comment:
Should we specify the Xcode version here?  10.0 vs 10.1?

Reply:
I think I'd lean towards 10.1, the 10.1 GM came out only a month after 10.0 GM and the minimum macOS version required is the same.